src/ucp/rndv/rndv.c

```diff
@@ -177,13 +169,20 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_proto_progress_rndv_rtr, (self),
 
     /* send the RTR. the pack_cb will pack all the necessary fields in the RTR */
     packed_rkey_size = ucp_ep_config(rndv_req->send.ep)->rndv.rkey_size;
-    status = ucp_do_am_single(self, UCP_AM_ID_RNDV_RTR, ucp_rndv_rtr_pack,
-                              sizeof(ucp_rndv_rtr_hdr_t) + packed_rkey_size);
-    if (status == UCS_OK) {
-        /* release rndv request */
+    status           = ucp_do_am_single(self, UCP_AM_ID_RNDV_RTR,
+                                        ucp_rndv_rtr_pack,
+                                        sizeof(ucp_rndv_rtr_hdr_t) +
+                                        packed_rkey_size);
+    if (ucs_unlikely((status != UCS_OK) && (status != UCS_ERR_NO_RESOURCE))) {
+        ucp_request_send_untrack(rndv_req);
         ucp_request_put(rndv_req);
     }
 
+    /* don't release rndv request in case of success, since it was sent to
```

Comment:
maybe start tracking here instead of `ucp_rndv_req_send_rtr`? Then would not need to untrack on failure